src/main/java/service/operation/PurchaseHandler.java

```diff
@@ -0,0 +1,13 @@
+package service.operation;
+
+import db.Storage;
+import models.FruitTransaction;
+
+public class PurchaseHandler implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction fruitTransaction) {
+        Integer oldQuantity = Storage.fruitMap.get(fruitTransaction.getFruit());
+        Storage.fruitMap.put(fruitTransaction.getFruit(),
+                oldQuantity - fruitTransaction.getQuantity());

```

Comment:
let's check if the product quantity in storage is enough and throw an exception if needed.